packages/dd-trace/src/config.js

```diff
@@ -97,39 +97,49 @@ function validateOtelPropagators (propagators) {
   }
 }
 
-function validateEnvVarType (envVar) {
-  const value = getEnvironmentVariable(envVar)
+/**
+ * Validate the type of an environment variable
+ * @param {string} envVar - The name of the environment variable
+ * @param {string} [value] - The value of the environment variable
+ * @returns {boolean} - True if the value is valid, false otherwise
+ */
+function isInvalidOtelEnvironmentVariable (envVar, value) {
+  // Skip validation if the value is undefined (it was not set as environment variable)
+  if (value === undefined) return false
+
   switch (envVar) {
     case 'OTEL_LOG_LEVEL':
-      return VALID_LOG_LEVELS.has(value)
+      return !VALID_LOG_LEVELS.has(value)
     case 'OTEL_PROPAGATORS':
     case 'OTEL_RESOURCE_ATTRIBUTES':
     case 'OTEL_SERVICE_NAME':
-      return typeof value === 'string'
+      return typeof value !== 'string'
     case 'OTEL_TRACES_SAMPLER':
-      return getFromOtelSamplerMap(value, getEnvironmentVariable('OTEL_TRACES_SAMPLER_ARG')) !== undefined
+      return getFromOtelSamplerMap(value, getEnv('OTEL_TRACES_SAMPLER_ARG')) === undefined
     case 'OTEL_TRACES_SAMPLER_ARG':
-      return !Number.isNaN(Number.parseFloat(value))
+      return Number.isNaN(Number.parseFloat(value))
     case 'OTEL_SDK_DISABLED':
-      return value.toLowerCase() === 'true' || value.toLowerCase() === 'false'
+      return value.toLowerCase() !== 'true' && value.toLowerCase() !== 'false'
     case 'OTEL_TRACES_EXPORTER':
     case 'OTEL_METRICS_EXPORTER':
     case 'OTEL_LOGS_EXPORTER':
-      return value.toLowerCase() === 'none'
+      return value.toLowerCase() !== 'none'
     default:
-      return false
+      return true
   }
 }
 
 function checkIfBothOtelAndDdEnvVarSet () {
   for (const [otelEnvVar, ddEnvVar] of Object.entries(otelDdEnvMapping)) {
-    if (ddEnvVar && getEnvironmentVariable(ddEnvVar) && getEnvironmentVariable(otelEnvVar)) {
+    const otelValue = getEnv(otelEnvVar)
+
+    if (ddEnvVar && getEnv(ddEnvVar) && otelValue) {
```

Comment:
Nit, to not call `getEnv` unless there's a truthy `otelValue`:

```suggestion
    if (otelValue && ddEnvVar && getEnv(ddEnvVar)) {
```

Reply:
I'll open a follow-up PR for it soon